hardware/BleBox.cpp

```diff
@@ -586,14 +569,9 @@ namespace http {
 			}
 
 			std::string hwid = request::findValue(&req, "idx");
-			if (hwid == "")
-				return;
-			int iHardwareID = atoi(hwid.c_str());
-			CDomoticzHardwareBase *pBaseHardware = m_mainworker.GetHardware(iHardwareID);
+			CDomoticzHardwareBase *pBaseHardware = GetHardwareBase(hwid, HTYPE_BleBox);
```

Comment:
Plz use the normal mainworker gethardware function (with the ID), then after check if the TYPE is correct
You can then remove the webserver changes (which also gave build issues)